### PR TITLE
Configurable max procs max atoms master

### PR DESCRIPTION
**Pull request by michaelklishin**
## Proposed Changes

Same as #1528 by @gerhard but against `master`.

### CODE REVIEW

File: scripts/rabbitmq-env [hunk 1]
```diff
@@ -87,8 +87,14 @@ DEFAULT_SCHEDULER_BIND_TYPE="db"
 DEFAULT_DISTRIBUTION_BUFFER_SIZE=128000
 [ "x" = "x$RABBITMQ_DISTRIBUTION_BUFFER_SIZE" ] && RABBITMQ_DISTRIBUTION_BUFFER_SIZE=${DEFAULT_DISTRIBUTION_BUFFER_SIZE}
 
+DEFAULT_MAX_NUMBER_OF_PROCESSES=1048576
+[ "x" = "x$RABBITMQ_MAX_NUMBER_OF_PROCESSES" ] && RABBITMQ_MAX_NUMBER_OF_PROCESSES=="${DEFAULT_MAX_NUMBER_OF_PROCESSES}"
+
+DEFAULT_MAX_NUMBER_OF_ATOMS=5000000
+[ "x" = "x$RABBITMQ_MAX_NUMBER_OF_ATOMS" ] && RABBITMQ_MAX_NUMBER_OF_ATOMS=="${DEFAULT_MAX_NUMBER_OF_ATOMS}"
```
Comment (michaelklishin) [inline scripts/rabbitmq-env]: `==` seems to be bash-specific and is not available in zsh.

Comment (lukebakken) [reply scripts/rabbitmq-env]: I totally missed that, assignment vs comparison

**Comment by michaelklishin** [comment]
Another thing I've noticed, which doesn't have to be addressed in this PR, is that in `rabbitmq-env.conf` environment variables are not supposed to have `RABBITMQ_` prefixes. I don't think it works that way for all variables, trying to prove myself wrong a.t.m.

File: scripts/rabbitmq-env (hunk 1 above)
Comment (michaelklishin) [reply scripts/rabbitmq-env]: ShellCheck catches it, too:

```
In scripts/rabbitmq-env line 91:
[ "x" = "x$RABBITMQ_MAX_NUMBER_OF_PROCESSES" ] && RABBITMQ_MAX_NUMBER_OF_PROCESSES=="${DEFAULT_MAX_NUMBER_OF_PROCESSES}"
                                                                                  ^-- SC1097: Unexpected ==. For assignment, use =. For comparison, use [/[[.


In scripts/rabbitmq-env line 94:
[ "x" = "x$RABBITMQ_MAX_NUMBER_OF_ATOMS" ] && RABBITMQ_MAX_NUMBER_OF_ATOMS=="${DEFAULT_MAX_NUMBER_OF_ATOMS}"
                                                                          ^-- SC1097: Unexpected ==. For assignment, use =. For comparison, use [/[[.
```

**Comment by lukebakken** [comment]
@michaelklishin that's correct and is also something I just noticed. Addressing it as well.

**Comment by gerhard** [comment]
@lukebakken [this comment](https://github.com/rabbitmq/rabbitmq-server/pull/1528#issuecomment-368149884) applies here as well. Pretty sure the original env vars which started with `RABBITMQ_` were correct.

**Comment by michaelklishin** [comment]
[The docs](http://www.rabbitmq.com/configure.html#define-environment-variables) suggest otherwise.

**Comment by michaelklishin** [comment]
Merging per verbal approval from @dumbbell.

**Pull request merged**